crates/fuel-core/src/schema/tx.rs

```diff
@@ -862,4 +880,40 @@ impl<'a> ContextExt for Context<'a> {
             }
         }
     }
+
+    async fn estimate_predicates(
+        &self,
+        mut tx: FuelTx,
+        query: impl PredicateStorageRequirements + Send + Sync + 'static,
+    ) -> anyhow::Result<FuelTx> {
+        let mut has_predicates = false;
+
+        for input in tx.inputs()? {
+            if input.predicate().is_some() {
+                has_predicates = true;
+                break;
+            }
+        }
+
+        if !has_predicates {
+            return Ok(tx);
+        }
```

Comment:
maybe something like:

```suggestion
        if tx.inputs()?.all(|input| input.predicate().is_none()) {
            return Ok(tx);
        }
```